module/zfs/zfs_ioctl.c

```diff
@@ -6861,6 +7111,17 @@ zfsdev_ioctl(struct file *filp, unsigned cmd, unsigned long arg)
 		break;
 	}
 
+	/*
+	 * Ensure that all input pairs are valid before we pass down to
+	 * the lower layers. The vectored functions can use fnvlist_lookup
+	 * since zfs_check_input_nvpairs() confirms that required pairs
+	 * exist and are of the correct type.
```

Comment:
As I understand it the vectored functions can use `fnvlist_lookup()` only for required keys, optional ones can't take advantage of this.  Let's make that perfectly clear.

Reply:
fixed